File: moon_upload_img_base/src/main/java/com/meili/component/uploadimg/common/IRetryHandler.java
package com.meili.component.uploadimg.common;

import java.io.Serializable;

/**
 * 重试处理器
 * Created by imuto on 17/12/21.
 */
public interface IRetryHandler extends Serializable {
    /** 是否可以重试 */
    boolean retry(Throwable throwable);

    /** 重置计数等 */
    void reset();
}

File: moon_sdk_kit/src/main/java/com/meili/moon/sdk/http/IRetryHandler.kt
package com.meili.moon.sdk.http

/**
 * 重试的处理接口
 * Created by imuto on 17/11/28.
 */
interface IRetryHandler<in ReqIns> {
    /**
     * 是否可以重试
     * @param requestInstance 请求重试的实体
     * @param throwable 失败时发生的错误
     * @return true可以重试，false不能重试
     */
    fun retry(requestInstance: ReqIns, throwable: Throwable): Boolean

    /**获取最大重试次数*/
    fun getMaxTimes(): Int

    /**获取当前重试次数*/
    fun getTimes(): Int
}